app/helpers/validation_helper.php

```diff
@@ -0,0 +1,21 @@
+<?php
+function validate_between($str, $min, $max)
+{
+    $length = mb_strlen($str); // count characters based on encoding
+    return $length <= $max && $length >= $min;
+}
+
+function validate_username($str)
+{
+    return preg_match('/^[[:alnum:]_]*$/', $str);
+}
+
+function validate_unique_name($username)
+{
+    return !User::getByUsername($username);
```

Comment:
instead of creating a validation helper for unique username, why not make username in user table unique then just catch the SimpleDBI Exception for it po?